utils/db.ts

```diff
@@ -31,75 +30,106 @@ async function getValues<T>(
 }
 
 // Item
-interface InitItem {
+export interface Item {
   userId: string;
   title: string;
   url: string;
-}
-
-export interface Item extends InitItem {
+  // The below properties can be automatically generated upon item creation
   id: string;
   createdAt: Date;
   score: number;
 }
 
-export async function createItem(initItem: InitItem) {
-  const item: Item = {
+export function newItemProps(): Omit<Item, "userId" | "title" | "url"> {
+  return {
     id: crypto.randomUUID(),
     score: 0,
     createdAt: new Date(),
-    ...initItem,
   };
+}
 
-  const itemKey = ["items", item.id];
-  const itemByTime = ["items_by_time", item.createdAt.getTime(), item.id];
+/**
+ * Sets an item in KV.
+ *
+ * @example New item creation
+ * ```ts
+ * import { newItemProps, setItem, incrementAnalyticsMetricPerDay } from "@/utils/db.ts";
+ *
+ * const item: Item = {
+ *   userId: "example-user-id",
+ *   title: "example-title",
+ *   url: "https://example.com"
+ *   ..newItemProps(),
+ * };
+ *
+ * await setItem(item);
+ * await incrementAnalyticsMetricPerDay("items_count", item.createdAt);
+ * ```
+ */
+export async function setItem(item: Item) {
+  const itemsKey = ["items", item.id];
+  const itemsByTimeKey = ["items_by_time", item.createdAt.getTime(), item.id];
   const itemsByUserKey = ["items_by_user", item.userId, item.id];
 
   const res = await kv.atomic()
-    .check({ key: itemKey, versionstamp: null })
-    .check({ key: itemByTime, versionstamp: null })
-    .check({ key: itemsByUserKey, versionstamp: null })
-    .set(itemKey, item)
-    .set(itemByTime, item)
+    .set(itemsKey, item)
+    .set(itemsByTimeKey, item)
     .set(itemsByUserKey, item)
     .commit();
 
```

Comment:
Just wondering why the `.check` is required before but unnecessary now?

Reply:
Good point. I've added checks and renamed the function to `createItem()` as it's only ever used to create new items.